STYLE.md

```diff
@@ -0,0 +1,432 @@
+Haskell Style Guide
+============================
+> This style guide is mostly a copy of [Johan Tibell's guide](https://github.com/tibbe/haskell-style-guide/blob/master/haskell-style.md)
+> with some restructurization, elaboration on some topics and some additions.
+> This style guide's aims are code beauty, readability and understandability.
+
+General guide lines
+-------------------
+### Line Length
+
+Maximum line length is *80 characters* or *100 characters* if necessary.
```

Comment:
Now, I think we could further relax that restriction in one very specific case, namely:

1. Lines that end with long string literals.

The reason is that this allows using tabular layout in code that often gains from it a lot.

The relative downside is small -- after all, this is just prose, that overflows in constrained line scenarios, not code.

Reply:
I disagree here because it makes looking at changes to that code unsightly and deviates from the repo's overall format guidelines.

but I'd like to hear @erikd 's thoughts & @intricate 's thoughts.

Reply:
> I disagree here because it makes looking at changes to that code unsightly

First, I think it should be fair to note that this is not a general statement.
The default, non-side-by-side diff view on Github is 116 characters wide, and it does not overflow.  The cases that blow that character budget are sufficiently rare, it shouldn't be a big problem to scroll to the right.  As an example:

![Screenshot from 2019-08-30 15-26-42](https://user-images.githubusercontent.com/452652/64129945-fa1e8180-cdc7-11e9-9a58-588fb953e559.png)

Second, we should consider that the Github diff view isn't the main avenue for reading code, and especially so not for writing code.

So pessimising the rare cases that gain a lot from tabular layout by enforcing an absolute requirement on line length seems overly harsh to me.

Reply:
I do not think there should be an exception for long string literals, especially not when the start at column 40.

Reply:
The code @deepfire links is pretty close to a example of how I would *not* format that file.

Reply:
I can accept that style requirement, by regarding it as a decree, even though I fail to see the balance of arguments.

Let's do it!